vignettes/adtr.Rmd

```diff
@@ -0,0 +1,599 @@
+  ) %>%
+  derive_var_merged_summary(
+    dataset_add = adtr,
+    by_vars = vars(USUBJID, AVISIT),
+    filter_add = str_detect(PARAMCD, "DIAM[12345]") & ANL01FL == "Y",
+    new_var = LSASS,
+    analysis_var = TRLNKID,
+    summary_fun = function(x) paste(sort(x), collapse = ", ")
+  ) %>%
+  derive_var_merged_summary(
+    dataset_add = adtr,
+    by_vars = vars(USUBJID, AVISIT),
+    filter_add = str_detect(PARAMCD, "DIAM[12345]"),
+    new_var = ADY,
+    analysis_var = ADY,
+    summary_fun = function(x) min(x, na.rm = TRUE)
+  ) %>%
+  derive_var_merged_summary(
+    dataset_add = adtr,
+    by_vars = vars(USUBJID, AVISIT),
+    filter_add = str_detect(PARAMCD, "DIAM[12345]"),
+    new_var = ADT,
+    analysis_var = ADT,
+    summary_fun = function(x) min(x, na.rm = TRUE)
+  ) %>%
+  mutate(
+    ANL01FL = if_else(LSEXP == LSASS, "Y", NA_character_)
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  arrange(adtr_sum, USUBJID, AVISITN),
+  display_vars = vars(USUBJID, PARAMCD, AVISIT, LSEXP, LSASS, ADT, ADY)
+)
+```  
+
+## Derive Baseline (`BASETYPE`, `ABLFL`, `BASE`) {#base}
+
+These functions are available in `admiral`, below examples show its usage for `ADTR` 
+
+```{r}
+adtr_sum <- adtr_sum %>%
+  restrict_derivation(
+    derivation = derive_var_extreme_flag,
+    args = params(
+      by_vars = vars(USUBJID),
+      order = vars(ADY),
+      new_var = ABLFL,
+      mode = "last"
+    ),
+    filter = ADY <= 1
+  ) %>%
+  derive_var_base(
+    by_vars = vars(USUBJID)
+  )
+```
+
+## Derive `NADIR` {#nadir}
+
+Once we have a TSD parameter, the NADIR is set to the lowest sum of diameters in
+Analysis Value before the current observation, where Analysis Flag 01 =Y.  The
+first observations after baseline (depending on how you have defined baseline
+earlier) is set to analysis value where baseline record flag = Y
+
+```{r}
+adtr_sum <- adtr_sum %>%
+  derive_vars_joined(
+    dataset_add = adtr_sum,
+    by_vars = vars(USUBJID),
+    order = vars(AVAL),
+    new_vars = vars(NADIR = AVAL),
+    join_vars = vars(ADY),
+    filter_add = ADY > 0 & ANL01FL == "Y",
+    filter_join = ADY.join < ADY,
+    mode = "first",
+    check_type = "none"
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  arrange(adtr_sum, USUBJID, AVISITN),
+  display_vars = vars(USUBJID, PARAMCD, AVISIT, AVAL, NADIR)
+)
+```  
+
+## Derive Change from Baseline (`CHG`, `PCHG`) {#chg}
+
+These functions are available in admiral, below examples show its usage for
+`ADTR`.  We calculate the change from baseline and percentage change from
+baseline for the parameter TSD and the variable `NADIR`.
+
+```{r}
+adtr_sum <- adtr_sum %>%
+  derive_var_chg() %>%
+  derive_var_pchg() %>%
+  mutate(
+    CHGNAD = AVAL - NADIR,
+    PCHGNAD = 100 * CHGNAD / NADIR
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  arrange(adtr_sum, USUBJID, AVISITN),
+  display_vars = vars(USUBJID, AVISIT, AVAL, NADIR, CHG, CHGNAD, PCHGNAD)
```

Comment:
**[From Catherine Jin via Stephen Gormley]**

Can we add BASE and PCHG to the display?

Reply:
done